### PR TITLE
Support EVP_get_digestbyname binding

### PR DESCRIPTION
**Pull request by crab2313**
Fixes: #1391

### CODE REVIEW

File: openssl/src/hash.rs [hunk 1]
```diff
@@ -46,6 +47,18 @@ impl MessageDigest {
         }
     }
 
+    pub fn from_name(name: &str) -> Option<MessageDigest> {
```
Comment (sfackler) [inline openssl/src/hash.rs:50]: Can you add a quick bit of documentation like what from_nid has?

**Comment by sfackler** [comment]
LGTM other than adding the docs!

File: openssl/src/hash.rs (hunk 1 above)
Comment (crab2313) [reply openssl/src/hash.rs:55]: Sure.

**Comment by crab2313** [comment]
EVP_get_digestbyname should be called after OPENSSL_add_all_algorithms() with openssl 1.0.0.  The code is changed to fix the bug.

File: openssl/src/hash.rs [hunk 2]
```diff
@@ -46,6 +47,25 @@ impl MessageDigest {
         }
     }
 
+    /// Returns the `MessageDigest` corresponding to an algorithm name.
+    ///
+    /// This corresponds to [`EVP_get_digestbyname`].
+    ///
+    /// [`EVP_get_digestbyname`]: https://www.openssl.org/docs/man1.1.0/crypto/EVP_DigestInit.html
+    pub fn from_name(name: &str) -> Option<MessageDigest> {
+        #[cfg(not(ossl110))]
```
Comment (sfackler) [inline openssl/src/hash.rs:56]:
```suggestion
```

Init is already a no-op on versions that don't need it, so it can be called unconditionally.

**Pull request merged**